tests/foreman/cli/test_discoveredhost.py

```diff
@@ -14,50 +14,135 @@
 
 :Upstream: No
 """
-from time import sleep
-
 import pytest
+from wait_for import wait_for
 
-from robottelo.cli.base import CLIReturnCodeError
-from robottelo.cli.discoveredhost import DiscoveredHost
 
 pytestmark = [pytest.mark.run_in_one_thread]
 
 
-def _assertdiscoveredhost(hostname):
-    """Check if host is discovered and information about it can be
-    retrieved back
+@pytest.mark.e2e
+@pytest.mark.on_premises_provisioning
+@pytest.mark.parametrize('module_provisioning_sat', ['discovery'], indirect=True)
+@pytest.mark.parametrize('pxe_loader', ['bios', 'uefi'], indirect=True)
+@pytest.mark.rhel_ver_match('8')
+def test_positive_provision_pxe_host_cli(
+    module_provisioning_rhel_content,
+    module_discovery_sat,
+    provisioning_host,
+    provisioning_hostgroup,
+):
+    """Provision a pxe-based discovered host from CLI
 
-    Introduced a delay of 300secs by polling every 10 secs to get expected
-    host
-    """
-    for _ in range(30):
-        try:
-            discovered_host = DiscoveredHost.info({'name': hostname})
-        except CLIReturnCodeError:
-            sleep(10)
-            continue
-        return discovered_host
+    :id: 91bb254b-3c30-4608-b1ba-e18bcc22efb5
 
+    :parametrized: yes
 
-@pytest.mark.stubbed
-@pytest.mark.tier3
-def test_positive_pxe_based_discovery():
-    """Discover a host via PXE boot by setting "proxy.type=proxy" in
-    PXE default
+    :Setup: Provisioning and discovery should be configured
 
-    :id: 25e935fe-18f4-477e-b791-7ea5a395b4f6
-
-    :Setup: Provisioning should be configured
+    :Steps:
 
-    :Steps: PXE boot a host/VM
+        1. Boot up the host to discover
+        2. Provision the host
 
-    :expectedresults: Host should be successfully discovered
+    :expectedresults: Host should be provisioned successfully from CLI
 
     :CaseImportance: Critical
 
     :BZ: 1731112
     """
+    sat = module_discovery_sat.sat
+    provisioning_host.power_control(ensure=False)
+    mac = provisioning_host._broker_args['provisioning_nic_mac_addr']
+
+    wait_for(
+        lambda: sat.api.DiscoveredHost().search(query={'mac': mac}) != [],
+        timeout=240,
+        delay=20,
+    )
+    discovered_host = sat.api.DiscoveredHost().search(query={'mac': mac})[0]
+    discovered_host.hostgroup = provisioning_hostgroup
+    discovered_host.location = provisioning_hostgroup.location[0]
+    discovered_host.organization = provisioning_hostgroup.organization[0]
+    discovered_host.build = True
+    result = sat.cli.DiscoveredHost.provision(
+        {
+            'id': discovered_host.id,
+            'hostgroup-id': discovered_host.hostgroup.id,
+            'organization-id': discovered_host.organization.id,
+            'location-id': discovered_host.location.id,
+        }
+    )
+    assert 'Host created' in result[0]['message']
+    host = sat.api.Host().search(query={"search": f'id={discovered_host.id}'})[0]
+    assert host
```

Comment:
```suggestion
    assert sat.api.Host().search(query={"search": f'id={discovered_host.id}'})
```

Reply:
Good suggestion but I need the host var below in the code.

Reply:
Not sure, how this works for you, host variable will have a same build_status_label, and you're using it for wait_for which will use delay=10 to re-check this same status of Pending Install until the host is in Installed state.

Reply:
Not getting you here @Gauravtalreja1, in the wait_for block its host_entity.read.build_status.

Reply:
I can remove `host = host.read()` by changing it to `host.read().build_status_label == 'Installed '` after the wait_for block.